src/Dict/Dict.ts

```diff
@@ -7,6 +7,11 @@ export declare const keys: <T extends Record<string, unknown>>(
   dict: T,
 ) => ReadonlyArray<keyof T>
 export declare const merge: <A, B>(fst: A, snd: B) => A & B
+export declare const set: <T, K extends string, V>(
+  dict: T,
+  key: K,
+  value: V,
+) => T & { [key in K]: V }
```

Comment:
looks good as well! I think we should extend `T` with `string | number` and add tests for both cases 

```ts
export declare const set: <T, K extends string | number, V>(
  dict: T,
  key: K,
  value: V,
) => T & Record<K, V>
```